### ISSUE TITLE
Raising `AttributeError` instead of `KeyError` for unknown attributes.

### ISSUE THREAD

**Issue by BenRKarl** [googleapis/proto-plus-python#31]
Since wrapped messages store known fields in a `dict` a `KeyError` is raised when accessing unknown fields, for example when doing `getattr(request, 'foo')`. This also seems to prevent `hasattr` from returning a `bool` as it only handles `AttributeError`.

It seems more conventional for `AttributeError` to be raised here. There may be some context I'm missing as to why `KeyError` is raised so wanted to discuss further here.

**Comment by BenRKarl** [comment]
For context here are some examples. In `google-ads-python` [here](https://github.com/googleads/google-ads-python/blob/506cba4608acb6cee9efa8a6f06070a7eb51953d/google/ads/google_ads/interceptors/logging_interceptor.py#L120) we examine message objects to retrieve fields for logging purposes.

This would be preferred but it won't work because `KeyError` is raised instead of `AttributeError`
```python
if hasattr(request, 'customer_id'):
    # do stuff
```

Instead we've updated the code to catch `KeyError` as such:
```
try:
    getattr(request, 'customer_id'):
except KeyError:
    # do stuff
```

However some message that don't get wrapped by this library, such as LROs, do raise `AttributeError` so our current implementation is:

```
try:
    getattr(request, 'customer_id'):
except (KeyError, AttributeError):
    # do stuff
```

**Comment by lukesneeringer** [comment]
@software-dov If you agree, I think this is worth fixing (with a relatively simple `except KeyError as ex: raise AttributeError(str(ex))`.

**Comment by software-dov** [comment]
Yes, this is my assessment as well.

**Comment by software-dov** [comment]
Fixed by #73
